### PR TITLE
Create an InternalBatch proto which is a superset of Batch.

### PR DESCRIPTION
**Pull request by spencerkimball**
The internal batch allows internal commands (namely: push txn & resolve
intent).

### CODE REVIEW

File: proto/internal.proto [hunk 1]
```diff
@@ -181,7 +181,60 @@ message InternalLeaderLeaseResponse{
   optional ResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 }
 
+// An InternalRequestUnion contains exactly one of the optional requests.
```
Comment (tschottdorf) [inline proto/internal.proto]: Add a comment to the BatchRequest/BatchResponseUnion that whenever something is added there, it needs to be added to the two corresponding internal counterparts.

File: proto/internal.go [hunk 2]
```diff
@@ -50,3 +51,27 @@ func InternalTimeSeriesDataFromValue(value *Value) (*InternalTimeSeriesData, err
...
+	}
+	br.Requests = append(br.Requests, union)
+}
+
+// Add adds a response to the batch response.
+func (br *InternalBatchResponse) Add(reply Response) {
+	union := InternalResponseUnion{}
+	if !union.SetValue(reply) {
+		// TODO(tschottdorf) evaluate whether this should return an error.
+		log.Fatalf("unable to add %T to batch response", reply)
```
Comment (tschottdorf) [inline proto/internal.go]: s/batch/internal batch/g
also above

File: proto/internal.proto [hunk 3]
```diff
@@ -181,7 +181,60 @@ message InternalLeaderLeaseResponse{
...
+    GetRequest get = 2;
+    PutRequest put = 3;
+    ConditionalPutRequest conditional_put = 4;
+    IncrementRequest increment = 5;
+    DeleteRequest delete = 6;
+    DeleteRangeRequest delete_range = 7;
+    ScanRequest scan = 8;
+    EndTransactionRequest end_transaction = 9;
+
+    InternalPushTxnRequest internal_push_txn = 10;
```
Comment (tschottdorf) [inline proto/internal.proto]: maybe some more room here for good measure?

File: proto/internal.proto [hunk 4]
```diff
@@ -181,7 +181,60 @@ message InternalLeaderLeaseResponse{
...
+    GetResponse get = 2;
+    PutResponse put = 3;
+    ConditionalPutResponse conditional_put = 4;
+    IncrementResponse increment = 5;
+    DeleteResponse delete = 6;
+    DeleteRangeResponse delete_range = 7;
+    ScanResponse scan = 8;
+    EndTransactionResponse end_transaction = 9;
+
+    InternalPushTxnResponse internal_push_txn = 10;
```
Comment (tschottdorf) [inline proto/internal.proto]: ditto

**Comment by tschottdorf** [comment]
LGTM

File: proto/internal.go (hunk 2 above)
Comment (spencerkimball) [inline proto/internal.go]: Done.

File: proto/internal.proto (hunk 1 above)
Comment (spencerkimball) [inline proto/internal.proto]: Done.

File: proto/internal.proto (hunk 3 above)
Comment (spencerkimball) [inline proto/internal.proto]: Good idea. Done.

**Pull request merged**